File: runelite/runelite-client/src/main/java/net/runelite/client/plugins/gamedataserver/model/exchange/ExchangeSlotState.java
package net.runelite.client.plugins.gamedataserver.model.exchange;

import net.runelite.api.GrandExchangeOfferState;

public enum ExchangeSlotState {
	EMPTY,
	CANCELLED_BUY,
	BUYING,
	BOUGHT,
	CANCELLED_SELL,
	SELLING,
	SOLD;

	public static ExchangeSlotState fromGrandExchangeOfferState(GrandExchangeOfferState state) {
		return ExchangeSlotState.valueOf(state.name());
	}
}

File: core/core/clients/gds/models/exchange/exchange_slot_state.py
from enum import Enum, auto


class ExchangeSlotState(Enum):

    NOT_SPECIFIED = auto()
    EMPTY = auto()
    CANCELLED_BUY = auto()
    BUYING = auto()
    BOUGHT = auto()
    CANCELLED_SELL = auto()
    SELLING = auto()
    SOLD = auto()

    @classmethod
    def from_str(cls, state: str) -> "ExchangeSlotState":
        try:
            return cls[state.upper()]
        except KeyError:
            return cls.NOT_SPECIFIED
